examples/distributed_tree/distributed_knn.cpp

```diff
@@ -0,0 +1,93 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <cstdarg>
+#include <cstdio>
+#include <iostream>
+#include <random>
+#include <vector>
+
+#include <mpi.h>
+
+using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+using MemorySpace = ExecutionSpace::memory_space;
+
+namespace Example
+{
+template <class Points>
+struct Nearest
+{
+  Points points;
+  int k;
+};
+template <class Points>
+Nearest(Points const &, int) -> Nearest<Points>;
```

Comment:
I almost feel like we should provide this facility in core ArborX, together with wrapping points with a radius. We use it often, and have to repeat every time.

Reply:
I had similar thoughts